### PR TITLE
[SPARK-48680][SQL][DOCS] Add missing Java APIs and language-specific documentations for char/varchar

### PR DESCRIPTION
**Pull request by yaooqinn**
### What changes were proposed in this pull request?

- Add Java APIs to create char/varchar types
- Doc char/varchar to language-specific tables

### Why are the changes needed?
- feature parity among polyglot APIs
- document improvement

### Does this PR introduce _any_ user-facing change?
yes, new APIs

### How was this patch tested?

doc build
### Was this patch authored or co-authored using generative AI tooling?
no

### CODE REVIEW

File: docs/sql-ref-datatypes.md [hunk 1]
```diff
@@ -157,6 +159,8 @@ You can access them by doing
 |**DoubleType**|Double|DoubleType|
 |**DecimalType**|java.math.BigDecimal|DecimalType|
 |**StringType**|String|StringType|
+|**CharType(length)**|String|CharType(length)|
+|**VarcharType(length)**|String|VarcharType(length)|
```
Comment (cloud-fan) [inline docs/sql-ref-datatypes.md:163]: I'm a bit hesitant to document the,. They are not real data types yet. Once read from the table, they become StringType with special annotations.

Comment (yaooqinn) [reply docs/sql-ref-datatypes.md:163]: Please take a look at L40

File: docs/sql-ref-datatypes.md [hunk 2]
```diff
@@ -188,6 +192,8 @@ please use factory methods provided in
 |**DoubleType**|double or Double|DataTypes.DoubleType|
 |**DecimalType**|java.math.BigDecimal|DataTypes.createDecimalType()<br/>DataTypes.createDecimalType(*precision*, *scale*).|
 |**StringType**|String|DataTypes.StringType|
+|**CharType(length)**|String|Not supported|
+|**VarcharType(length)**|String|Not supported|
```
Comment (cloud-fan) [inline docs/sql-ref-datatypes.md:196]: shall we add the corresponding Java API?

Comment (yaooqinn) [reply docs/sql-ref-datatypes.md]: Addressed. Thank you

**Review by cloud-fan — APPROVED** [review]

**Comment by yaooqinn** [comment]
Merged to master.

Thank you @cloud-fan for the review

**Pull request closed without merging**